synapseclient/models/dataset.py

```diff
@@ -0,0 +1,417 @@
+import dataclasses
+from collections import OrderedDict
+from copy import deepcopy
+from dataclasses import dataclass, field
+from datetime import date, datetime
+from typing import Any, Dict, List, Optional, Union
+
```

Comment:
Brad will be finishing out the Dataset in a follow-up pull request. It's in an incomplete state at the moment and is only partially functional.